3rdparty/snmalloc/allocator.cpp

```diff
@@ -22,8 +22,7 @@ void* ThreadAllocUntyped::get()
 #define OPEN_ENCLAVE
 #define SNMALLOC_SGX
 #define USE_RESERVE_MULTIPLE 1
-#define IS_ADDRESS_SPACE_CONSTRAINED
-#define SNMALLOC_EXTERNAL_THREAD_ALLOC
```

Comment:
```C++
#define SNMALLOC_EXTERNAL_THREAD_ALLOC
```
Is still required.  This stop the allocator using C++ thread local state.  This should fix the build errors.